### ISSUE TITLE
Support VF interrupt binding to specified CPU

### ISSUE THREAD

**Issue by hustcat** [hustcat/sriov-cni#10]
CNI spec support [ordered plugin chaining](https://github.com/containernetworking/cni/pull/346), we can write a interrupt binding plugin to bind interrupt of VF to assigned CPU. @rkamudhan

**Comment by rkamudhan** [comment]
yes, that is great idea to work on.

**Comment by rkamudhan** [comment]
@hustcat Can we have a discussion on this implementation ?

**Comment by lmdaly** [comment]
@hustcat @rkamudhan We are currently exploring this feature in our PoC - maybe we can collaborate together on it?

**Comment by hustcat** [comment]
Yeah, @lmdaly @rkamudhan Do you have any detailed implementation?

**Comment by hustcat** [comment]
We should think how to coordinate with k8s, and k8s don't support cpuset for container now. Ref k8s  [#10570](https://github.com/kubernetes/kubernetes/issues/10570).

**Comment by lmdaly** [comment]
@hustcat the new native CPU Manager in Kubernetes uses cpusets to assigns cores at a container level. It should be an alpha feature in K8s 1.8.
Here is the proposal doc: https://github.com/ConnorDoyle/community/blob/a21b6adf7382ccf88a0ca42b133599af39eb4dd0/contributors/design-proposals/cpu-manager.md
and the code:
https://github.com/kubernetes/kubernetes/pull/49186

**Comment by ConnorDoyle** [comment]
Here's the stable link to the proposal (it has been merged): [CPU manager](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/cpu-manager.md)

**Comment by ConnorDoyle** [comment]
Here's a general top-level issue to help track work (since we need more than just coordination with CNI): https://github.com/kubernetes/kubernetes/issues/49964
